src/structure_managers/adaptor_neighbour_list.hh

```diff
@@ -628,28 +637,30 @@ namespace rascal {
 
       if (Order < (traits::MaxOrder - 1)) {
         return IncreaseHelper_t::get_neighbour_atom_tag(this->manager, cluster,
-                                                        index);
+                                                        iteration_index);
       } else {
         auto && offset = this->offsets[cluster.get_cluster_index(Layer)];
-        return this->neighbours_atom_tag[offset + index];
+        return this->neighbours_atom_tag[offset + iteration_index];
       }
     }
 
     //! Returns atom type given an atom tag, also works for ghost atoms
     inline int & get_atom_type(const int & atom_tag) {
+      // return this->atom_types[this->get_atom_index(atom_tag)];
       return this->atom_types[atom_tag];
     }
 
     /* If consider_ghost_neighbours=true and the atom tag corresponds to an
      * ghost atom, then it returns it cluster index of the atom in the original
      * cell.
      */
-    size_t get_atom_index(const int atom_tag) const {
+    inline size_t get_atom_index(const int atom_tag) const {
       return this->atom_index_from_atom_tag_list[atom_tag];
     }
 
     //! Returns the type of a given atom, given an AtomRef
     inline const int & get_atom_type(const int & atom_tag) const {
+      // return this->atom_types[this->get_atom_index(atom_tag)];
```

Comment:
can this go away? or should this be the version which is actually correct?

Reply:
this is where I am not sure about the design. here like at other places atom_tag is used to access arrays, these arrays are designed to work like these so no problem should arise but there might be a better way to do it. What do you think ?

Reply:
Ok, then we leave it for now. Maybe adding a comment line why we want to keep the design around as a reminder for now?